moshi/moshi/quantization/core_vq.py

```diff
@@ -300,6 +404,14 @@ def forward(self, x: torch.Tensor, initialize: bool = True) -> _VQForwardResult:
         x = self._rearrange_input(x)
         quantized, codes, metrics = self._codebook(x, initialize=initialize)
 
+        if self.training:
+            quantized = x + (quantized - x).detach()
+
+        if self.training:
+            loss = F.mse_loss(x, quantized.detach())
+        else:
+            loss = zero_scalar(x.device)
+
         loss = zero_scalar(x.device)
```

Comment:
Hi! Shouldn't this line be removed now that we handle it in lines 410-413?

```suggestion
```

Reply:
great catch !